model/project_ref.go

```diff
@@ -1141,21 +1141,27 @@ func GetTasksWithOptions(projectName string, taskName string, opts GetProjectTas
 }
 
 func FindFirstProjectRef() (*ProjectRef, error) {
-	projectRef := &ProjectRef{}
+	projectRefSlice := []ProjectRef{}
 	pipeline := projectRefPipelineForValueIsBool(ProjectRefPrivateKey, RepoRefPrivateKey, false)
-	pipeline = append(pipeline, bson.M{"$sort": "-" + ProjectRefDisplayNameKey}, bson.M{"$limit": 1})
+	pipeline = append(pipeline, bson.M{"$sort": bson.M{ProjectRefPrivateKey: -1}}, bson.M{"$limit": 1})
```

Comment:
Why did you switch what this is sorting by? It seems to me like the issue in the ticket is the nil pointer reference that you fixed by checking the length of the slice, but we still want to get the first alphabetical project.

Reply:
🤦 This is fixed. I did not mean to change the key. Good catch!

To be clear, the original issue wasn't the nil pointer reference, the issue was that "-DisplayName" is not valid mongodb bson, but the old driver used to allow it.

Fixing things for the new driver, we also had to switch to using an array since the new driver requires an array for the "out parameter" of the "Aggregate" function, which is also why I added the bounds checking.

Reply:
Ah gotcha, cool!